torch/_dynamo/variables/torch.py

```diff
@@ -1211,7 +1214,7 @@ def speculate_branch(branch):
             example_value = deepcopy_to_fake_tensor(example_res, tx.fake_mode)
 
             p_args = (lowered_node,) + p_args
-        elif self.value.__name__ == "wrap":
+        elif self.value.__name__ in ("wrap", "wrap_activation_checkpoint"):
```

Comment:
(no action required) There should be a check for side effects somewhere (that is, if the function passed to checkpoint or wrap mutates global state, then we have a problem). There isn't yet because we haven't added it because it is kind of complicated. Since wrap_activation_checkpoint is hidden behind a feature flag, I'm OK with this happening later (it's somewhere on the list of followups for wrap, https://github.com/pytorch/pytorch/issues/100278)